src/components/ServiceRequestForm.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import Categories from "./services/categories.json";
+import * as Yup from "yup";
+import ErrorMsg from "./errormsg";
+import FormContainer from './FormContainer';
+
+const getSubCategories = (categories, categoryId) => {
+	var category = categories.find(category => category.id === categoryId);
+	return category ? category.types : [];
+};
+
+const ServiceRequestForm = props => {
+
+	const [categories] = useState(Categories);
+	const [subCategories, setSubCategories] = useState([]);
+
+	const handleServiceRequestChange = (changeEvent, setFieldValue) => {
+		const { value } = changeEvent.currentTarget;
+		const subCategories = getSubCategories(categories, parseInt(value));
+		setFieldValue(changeEvent.currentTarget.name, changeEvent.currentTarget.value)
+		setFieldValue('subRequestType', '')
+		setSubCategories(subCategories);
+	};
+
+
+	return (
+
+		<React.Fragment>
+			<FormContainer title="How Can We Help?">
+				<Formik
+					initialValues={{
+						requestType: '',
+						subRequestType: '',
+					}}
+					validationSchema={Yup.object().shape({
+						requestType: Yup.string().required('Request Category is required'),
+						subRequestType: Yup.string().required('Sub Category is required'),
+					})}
+
+					onSubmit={(values, { setSubmitting }) => {
+
+						alert(JSON.stringify(values, null, 2));
+						setSubmitting(false);
+
+					}}
+				>
+					{
+						(props) => {
+							const { values, isSubmitting, errors, touched } = props;
+
+							return (
+
+								<Form >
+									<label htmlFor="requestType"
+										className={
+											errors.requestType && touched.requestType ? "input-feedback" : "text-label"}
+
+									>Request Category</label>
+									<br />
+									<Field
+										component="select"
+										name="requestType"
+										onChange={e => { handleServiceRequestChange(e, props.setFieldValue) }}
+										className={errors.requestType && touched.requestType ? "text-select error" : "text-select"}
```

Comment:
you want to avoid duplicating logic when possible. in this case, you are repeating the class 'text-select' in your conditional. You can write something like:

```jsx
className={`text-select ${errors.requestType && touched.requestType ? "error" : ""}`}
```

or use something like [classnames](https://github.com/JedWatson/classnames#readme)

Reply:
removed duplicating logic

Reply:
Verified duplicate code has been removed